### PR TITLE
Create main settings form 

### PR DESCRIPTION
**Pull request by Matveycom**

### CODE REVIEW

File: src/Form/HybridauthAdminSettings.php [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+  /**
+   * The module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * Constructs a new XmlSitemapCustomAddForm object.
```
Comment (ApacheEx) [inline src/Form/HybridauthAdminSettings.php]: let's fix description

File: src/Form/HybridauthAdminSettings.php [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    $form['fset_fields'] = [
+      '#type' => 'details',
+      '#title' => t('Required information'),
+      '#group' => 'vtabs',
+    ];
+    $form['fset_fields']['hybridauth_required_fields'] = [
+      '#type' => 'checkboxes',
+      '#title' => t('Required information'),
+      '#options' => [
+        'email' => t('Email address'),
```
Comment (VladBo) [inline src/Form/HybridauthAdminSettings.php]: $this->t()

**Review by VladBo — CHANGES_REQUESTED** [review]
$this->t()

File: src/Form/HybridauthAdminSettings.php [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+      '#default_value' => $values['hybridauth_email_verification_subject'],
+      '#maxlength' => 180,
+    ];
+    $form['fset_account']['fset_email_verification_template']['hybridauth_email_verification_body'] = [
+      '#type' => 'textarea',
+      '#title' => t('Body'),
+      '#default_value' => $values['hybridauth_email_verification_body'],
+      '#rows' => 12,
+    ];
+    $module_exist_token = $this->moduleHandler->moduleExists('token');
```
Comment (ApacheEx) [inline src/Form/HybridauthAdminSettings.php]: let's rename to `$token_exists`

File: src/Form/HybridauthAdminSettings.php [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+      '#type' => 'checkbox',
+      '#title' => t('Ask user for a password when registering'),
+      '#description' => t('Ask users to set password for account when registering through HybridAuth.'),
+      '#default_value' => $values['hybridauth_registration_password'],
+    ];
+    $form['fset_account']['hybridauth_override_realname'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Override Real name'),
+      '#default_value' => $values['hybridauth_override_realname'],
+      '#disabled' => !$this->moduleHandler->moduleExists('realname'),
```
Comment (ApacheEx) [inline src/Form/HybridauthAdminSettings.php]: let's remove support of realname at all

File: src/Form/HybridauthAdminSettings.php [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+      $form['fset_other']['fset_token']['hybridauth_token_tree'] = [
+        '#theme' => 'token_tree',
+        '#token_types' => [
+          'user',
+        ],
+        '#global_types' => TRUE,
+        '#dialog' => TRUE,
+      ];
+    }
+    $options = [
```
Comment (ApacheEx) [inline src/Form/HybridauthAdminSettings.php]: useless - let's use it without variable

File: src/Form/HybridauthAdminSettings.php [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+   */
+  public function getFormId() {
+    return 'hybridauth_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config = $this->config('hybridauth.settings');
```
Comment (ApacheEx) [inline src/Form/HybridauthAdminSettings.php:58]: let's have a config for it

**Review by ApacheEx — CHANGES_REQUESTED** [review]

**Review by ApacheEx — APPROVED** [review]

**Pull request merged**